### PR TITLE
Fetch archetype form `trailpack-****` except of `trailsjs/trails`

### PR DESCRIPTION
**Pull request by konstantinzolotarev**
Need to review some lines. Please take a look into coments before merge

### CODE REVIEW

File: src/app/steps/writing.js [hunk 1]
```diff
@@ -12,9 +12,21 @@ export default {
...
   serverDependentApi () {
-    let server = this.answers['web-engine']
-    this.fs.copy(path.resolve(TRAILS_TEMPLATE, 'api/controllers', server, '**'), this.destinationPath('api/controllers'))
-    this.fs.copy(path.resolve(TRAILS_TEMPLATE, 'api/policies', server, '**'), this.destinationPath('api/policies'))
+    const server = this.answers['web-engine']
+    let SERVER_TEMPLATE = null;
+    try {
+      SERVER_TEMPLATE = path.dirname(require.resolve('trailpack-' + server));
+
+      // TODO: may be copy everything form `api` ?
```
Comment (konstantinzolotarev) [inline src/app/steps/writing.js]: What do you think about this ?

File: src/app/steps/writing.js [hunk 2]
```diff
@@ -12,9 +12,21 @@ export default {
...
+      this.fs.copy(path.resolve(SERVER_TEMPLATE, 'archetype', 'api/controllers', '**'), this.destinationPath('api/controllers'))
+      this.fs.copy(path.resolve(SERVER_TEMPLATE, 'archetype', 'api/policies', '**'), this.destinationPath('api/policies'))
+    } catch(e) {
+      // Nothing to copy need somehow to inform about this.
```
Comment (konstantinzolotarev) [inline src/app/steps/writing.js]: Need a better way to inform that generator does not support `trailpack-****` out of the box.

**Comment by konstantinzolotarev** [comment]
Also this PR relates to trailsjs/trailpack-hapi#7

**Comment by tjwebb** [comment]
@konstantinzolotarev is this only for the web servers, or will this apply to any trailpack with an archetype/ folder?

**Comment by jaumard** [comment]
As I see is only for web server @tjwebb, I ask on Slack if you want to add a new command like `yo trails:trailpack trailpack-gulp` and the generator will install the trailpack and also copy all archetype files. Like this all trailpack can set api/config files instead of copy them on `postinstall`script or just put it on the README.

**Comment by tjwebb** [comment]
@konstantinzolotarev and I talked about this in the office today. I merged this as-is, and we will develop a way for any trailpack to include/generate files into the main app at some point in the future.

**Pull request merged**